File: java/src/main/java/geospat/Point.java
package geospat;


public class Point {
    private double x_;
    private double y_;

    public Point(double x,double y){x_=x;y_=y;}
    public Point(int x,int y){x_=x;y_=y;}
    public Point(){}
    
    public double getX(){return x_;}
    public double getY(){return y_;}
    public void setX(double x){x_=x;}
    public void setY(double y){y_=y;}

    @Override
    public int hashCode() {
        int x=(int)x_ << 16;
        int y=(int)y_ ;
        return (x|y);
    }

    @Override
    public boolean equals(Object obj) {
        Point point=(Point)obj;
        return (point.x_==x_ && point.y_==y_);
    }
}

File: go/Point.go
package main

import "fmt"

type Point struct {
	x float64
	y float64
}

//---------------------------------------------------------------------------------------
func (thisPt *Point) Equal(other *Point) bool {
	return (thisPt.x == other.x && thisPt.y == other.y)
}

//---------------------------------------------------------------------------------------
func (thisPt *Point) SetX(x float64) {
	thisPt.x = x
}

//---------------------------------------------------------------------------------------
func (thisPt *Point) SetY(y float64) {
	thisPt.y = y
}

//---------------------------------------------------------------------------------------
func (thisPt *Point) GetX() float64 {
	return thisPt.x
}

//---------------------------------------------------------------------------------------
func (thisPt *Point) GetY() float64 {
	return thisPt.y
}

//---------------------------------------------------------------------------------------
func (thisPt *Point) SetXL(x uint64) {
	thisPt.x = float64(x)
}

//---------------------------------------------------------------------------------------
func (thisPt *Point) SetYL(y uint64) {
	thisPt.y = float64(y)
}

//---------------------------------------------------------------------------------------
func (thisPt *Point) Gethash() uint64 {
	xl := uint64(thisPt.x)
	return (xl<<32 | uint64(thisPt.y))
}

//---------------------------------------------------------------------------------------
func (thisPt *Point) GetWKTString() string {
	return fmt.Sprintf("POINT(%f %f)", thisPt.x, thisPt.y)
}
